src/libraries/System.Text.RegularExpressions/tests/Regex.MultipleMatches.Tests.cs

```diff
@@ -127,19 +132,23 @@ public static IEnumerable<object[]> Matches_TestData()
                 }
             };
 
-            yield return new object[]
+            // RegexOptions.RightToLeft is not supported with NonBacktracking
+            if (options != RegexHelpers.RegexOptionNonBacktracking)
```

Comment:
would it be easier to segregate these into their own enumerator? Matches_TestData_WithOptions and Matches_TestData_WithOptions_Backtracking maybe? then you can reason about what tests are in what by just looking at which enumerators are used, with no if's.

Reply:
Yes, will do.